### ISSUE TITLE
GPU Acceleration

### ISSUE THREAD

**Issue by IanCassTwo** [deepskydetail/AstroSharp#4]
Fantastic tool! Would it be feasible to use Tensorflow instead of Neuralnet? This should make it possible to use GPU to accelerate the processing.

**Comment by deepskydetail** [comment]
Thanks! Good question. I currently use tensorflow/keras in Rstudio when
training the neural network and convert it to neuralnet because it's easier
to create the GUI that way. It's something that may be doable in the
future, especially if I can find someone with Python skills (I have *much
better* R knowledge than Python) to help me out :)

At the very least, I think I can get it to work with multiple CPU cores,
which could make it significantly faster.

**Comment by klaussfreire** [comment]
Hi. I was thinking on embarking on a similar endeavor. If you don't mind sharing the sources I wouldn't mind trying to make it work with GPU acceleration in python. I was even thinking of trying to improve on it. I mostly develop in linux, but I'm sure I can dust off my window skills a bit to port it back to windows.

If you're interested in collaborating, lets talk.

**Comment by deepskydetail** [comment]
Yes, I'd be interested. Send me an email! I am pretty busy until about mid
September, but after that I think we could set up something. Thanks!

**Comment by klaussfreire** [comment]
Sorry, but I don't see your email. I made mine visible in my profile, feel free to contact me and we can talk.
